src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -1841,5 +1855,202 @@ string[] linkingTargetFields
+                source: "TEST_SOURCEA",
+                relationshipMap: null,
+                graphQLDetails: null,
+                restDetails: new(new SupportedHttpVerb[] { }, restPathForEntity, true)
+            );
+            entityMap.Add("EntityA", sampleEntity);
+
+            RuntimeConfig runtimeConfig = new(
+                Schema: "UnitTestSchema",
+                DataSource: new DataSource(DatabaseType: DatabaseType.MSSQL, "", new()),
+                Runtime: new(
+                    Rest: new(),
+                    GraphQL: new(),
+                    Host: new(null, null)
+                ),
+                Entities: new(entityMap)
+            );
+
+            if (exceptionExpected)
+            {
+                DataApiBuilderException dabException =
+                    Assert.ThrowsException<DataApiBuilderException>(() => RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig));
+                Assert.AreEqual(expectedExceptionMessage, dabException.Message);
+                Assert.AreEqual(expected: HttpStatusCode.ServiceUnavailable, actual: dabException.StatusCode);
+                Assert.AreEqual(expected: DataApiBuilderException.SubStatusCodes.ConfigValidationError, actual: dabException.SubStatusCode);
+            }
+            else
+            {
+                RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig);
+            }
+        }
+
+        /// <summary>
+        /// Test to validate that when multiple entities have the same custom rest path configured, we throw an exception.
+        /// </summary>
+        /// <param name="exceptionExpected">Whether an exception is expected as a result of test run.</param>
+        /// <param name="restPathForFirstEntity">Custom rest path to be configured for the first entity.</param>
+        /// <param name="restPathForSecondEntity">Custom rest path to be configured for the second entity.</param>
+        /// <param name="expectedExceptionMessage">The expected exception message.</param>
+        [DataTestMethod]
+        [DataRow(false, "restPathA", "restPathB", DisplayName = "Unique rest paths configured for entities pass config validation.")]
+        [DataRow(true, "restPath", "restPath", "The rest path: restPath specified for entity: EntityB is already used by another entity.",
+            DisplayName = "Duplicate rest paths configures for entities fail config validation.")]
+        [DataRow(true, null, "EntityA", "The rest path: EntityA specified for entity: EntityB is already used by another entity.",
+            DisplayName = "Rest path for an entity configured as the name of another entity fails config validation.")]
+        public void ValidateUniqueRestPathsForEntitiesInConfig(
+            bool exceptionExpected,
+            string restPathForFirstEntity,
+            string restPathForSecondEntity,
+            string expectedExceptionMessage = "")
```

Comment:
Maybe covered in a different pre-existing test: would the validation method raise an exception when the rest path of entityB matches the top level entity name of entityA?

i.e.
toplevel entity name: "entityA"
restpath: "entityA"

toplevel entity name: "entityB"
restpath: "entityA"

Reply:
Already covered here:

https://github.com/Azure/data-api-builder/blob/f0aba3cd06f8ed74bd0f85f8ac7d7cd11ea019a6/src/Service.Tests/Unittests/ConfigValidationUnitTests.cs#L2005